src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponSlice.lua

```diff
@@ -0,0 +1,42 @@
+--!strict
+local Reflex = require("Packages/reflex")
+local Immut = require("Packages/immut")
+local Table = require("src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponTables")
+
+export type BrickbattleWeaponStateActions = {
+    setStatState: (playerName: string, numb: number, stat: string) -> (),
+    removeStatState: (playerName: string, stat: string) -> (),
+}
+
+type Data = {
+    [string]: Table.BrickbattlePlayerTable
+}
+local initialState: Data = {}
+local BrickbattlePlayerStateSlice = Reflex.createProducer(initialState, {
+    setPlayerWeaponState = function(state, playerName: string,weapon: string, Data: any)
```

Comment:
this is not a good name for the action that it does. it's completely overriding a player's data for a given weapon. how about `setTotalProjectileStateForPlayer = function(state, playerName: string, weapon: string, projectileData: any)`